src/metabase/actions/execution.clj

```diff
@@ -162,12 +162,13 @@
 (defn execute-action!
   "Execute the given action with the given parameters of shape `{<parameter-id> <value>}."
   [action request-parameters]
-  (case (:type action)
-    :implicit
-    (execute-implicit-action action request-parameters)
-    (:query :http)
-    (execute-custom-action action request-parameters)
-    (throw (ex-info (tru "Unknown action type {0}." (name (:type action))) action))))
+  (let [stringified-params (update-keys request-parameters name)]
```

Comment:
It's strange to see a correction of the deserialization in the execution namespace. I would have expected it in the `defendpoint` definitions, even if it's duplicated among multiple endpoints.

Reply:
My thought was that functions should be as permissive as reasonably possible in their input

Reply:
This is complicated i think. The select on actions runs a normalization over parameters which keywordizes the param keys. I think ever keywordizing what are really strings is the ultimate culprit of this behavior. The ids are strings, they come specified as strings from the FE, etc. And 

```clojure
(defn normalize-parameters-list
  "Normalize `parameters` or `parameter-mappings` when coming out of the application database or in via an API request."
  [parameters]
  (or (mbql.normalize/normalize-fragment [:parameters] parameters)
      []))
``` 
ultimately that just keywordizes all keys in a map. Which i think is clunky.

Reply:
Yes, it's the keywordization of the keys in the request middleware that requires us to reverse it. But that's a quirk of request body deserialization that the execution namespace shouldn't need to worry about.